cse110-sp24-group35/journal
# Implement Local-First

We address the local-first mechanic through ServiceWorker

## Description

When user accesses one of the pages, we will install a ServiceWorker. At the same time, the worker will actively try to cache all necessary links - including other html pages, all internal scripts, all vendor scripts (e.g. markdown renderer and nanostores), and css files. In this way, when user wants to navigate to other pages it'll pull straight from the worker so user will be able to freely use the journal app even without internet connection (the initial request requires internet connectivity, of course)

## Related Issue
Closes #13 





## How Has This Been Tested?
1. Launch test server
2. Go to any of HTML pages
3. Check installed service worker, and see requests that were being sent from the service worker
4. Refresh the page, you can see the html pages, scripts, and css styles have been cached by the ServiceWorker.


## Checklist
- [x] All automated tests have been passed locally (unit / e2e / linter / coverage)
- [x] New unit and E2E tests have been added to address the changes made
- [x] Manual testing has been performed according to the provided instructions
- [x] Code follows team's coding style, convention and standards
- [x] Documentation has been updated to reflect relevant changes (if applicable)

## Screenshots (if applicable):
